modules/terraform/options.go

```diff
@@ -102,3 +106,144 @@ func WithDefaultRetryableErrors(t *testing.T, originalOptions *Options) *Options
+// Returns an error if GetAllVariablesFromVarFileE returns an error, the key provided does not exist, or the value associated with the key is not a map
+func GetVariableAsMapFromVarFileE(t *testing.T, option *Options, fileName string, key string) (map[string]string, error) {
+	resultMap := make(map[string]string)
+	variables, err := GetAllVariablesFromVarFileE(t, option, fileName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	_, exists := variables[key]
+
+	if !exists {
+		return nil, InputFileKeyNotFound(key)
+	}
+
+	if reflect.TypeOf(variables[key]).String() != "[]map[string]interface {}" {
+		return nil, UnexpectedOutputType{Key: key, ExpectedType: "map", ActualType: reflect.TypeOf(variables[key]).String()}
+	}
+
+	mapKeys := variables[key].([]map[string]interface{})[0]
+
+	for mapKey, mapVal := range mapKeys {
+		resultMap[mapKey] = fmt.Sprintf("%v", mapVal)
+	}
+
+	return resultMap, nil
+}
+
+// GetVariableAsListFromVarFile Gets the string list represention of a variable from a provided input file found in VarFile
+func GetVariableAsListFromVarFile(t *testing.T, option *Options, fileName string, key string) []string {
+	result, err := GetVariableAsListFromVarFileE(t, option, fileName, key)
+	require.NoError(t, err)
+
+	return result
+}
+
+// GetVariableAsListFromVarFileE Gets the string list represention of a variable from a provided input file found in VarFile
+// Will return error if GetAllVariablesFromVarFileE returns an error, the key provided does not exist, or the value associated with the key is not a list
+func GetVariableAsListFromVarFileE(t *testing.T, option *Options, fileName string, key string) ([]string, error) {
+	resultArray := []string{}
+	variables, err := GetAllVariablesFromVarFileE(t, option, fileName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if _, exists := variables[key]; !exists {
+		return nil, InputFileKeyNotFound(key)
+	}
+
+	if reflect.TypeOf(variables[key]).String() != "[]interface {}" {
+		return nil, UnexpectedOutputType{Key: key, ExpectedType: "list", ActualType: reflect.TypeOf(variables[key]).String()}
+	}
+
+	for _, item := range variables[key].([]interface{}) {
+		resultArray = append(resultArray, fmt.Sprintf("%v", item))
+	}
+
+	return resultArray, nil
+}
+
+// GetAllVariablesFromVarFile Parses all data from a provided input file found in VarFile and returns them as a key-value map
+func GetAllVariablesFromVarFile(t *testing.T, option *Options, fileName string) map[string]interface{} {
+	variableMaps, err := GetAllVariablesFromVarFileE(t, option, fileName)
+	require.NoError(t, err)
+
+	return variableMaps
+}
+
+// GetAllVariablesFromVarFileE Parses all data from a provided input file found ind in VarFile and returns them as a key-value map
+// Retursn an error if the specified file does not exist, the specified file is not readable, or the specified file cannot be decoded from HCL
+func GetAllVariablesFromVarFileE(t *testing.T, option *Options, fileName string) (map[string]interface{}, error) {
```

Comment:
Why take in a `fileName` and `option` struct, and have to search for the file within it, rather than take in solely the path directly to a var file? This applies to all the methods in this PR.

Reply:
That makes sense. I think I wanted to use the options struct since I was originally thinking in just the scope of what is provided to options, but its just adding more unneeded logic. I'll adjust this.